File: todo-reservation-springboot/src/main/java/de/vkb/dojo/es/reservations/model/command/DeleteReservation.java
package de.vkb.dojo.es.reservations.model.command;

import com.fasterxml.jackson.annotation.*;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("delete-reservation")
public class DeleteReservation extends ReservationCommand {
    private final String operationId;
    private final String aggregateId;

    @JsonCreator
    public DeleteReservation(
            @JsonProperty("operationId") String operationId,
            @JsonProperty("aggregateId") String aggregateId
    ) {
        this.operationId = operationId;
        this.aggregateId = aggregateId;
    }

    @Override
    @JsonGetter
    public String getOperationId() {
        return operationId;
    }

    @Override
    @JsonGetter
    public String getAggregateId() {
        return aggregateId;
    }
}

File: todo-reservation-micronaut/src/main/kotlin/de/vkb/dojo/es/reservations/model/command/DeleteReservation.kt
package de.vkb.dojo.es.reservations.model.command

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonTypeName

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeName("delete-reservation")
data class DeleteReservation(
    override val operationId: String,
    override val aggregateId: String
): ReservationCommand
